target/linux/ath79/image/generic.mk

```diff
@@ -1340,6 +1340,23 @@ define Device/fortinet_fap-221-b
 endef
 TARGET_DEVICES += fortinet_fap-221-b
 
+define Device/fortinet_fap-221-c
+  $(Device/senao_loader_okli)
+  SOC := qca9558
+  DEVICE_VENDOR := Fortinet
+  DEVICE_MODEL := FAP-221-C
+  FACTORY_IMG_NAME := FP221C-9.99-AP-build999-999999-patch99
+  DEVICE_PACKAGES := ath10k-firmware-qca988x-ct kmod-ath10k-ct
+  IMAGE_SIZE := 20480k
+  LOADER_FLASH_OFFS := 0x0040000
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | \
+	  append-rootfs | pad-rootfs | \
+	  check-size | pad-to $$$$(IMAGE_SIZE) | \
+	  append-loader-okli-uimage $(1) | pad-to 10944k | \
+	  gzip-filename $$$$(FACTORY_IMG_NAME)
```

Comment:
the last `pad-to` must also be adjusted for the same reason that `IMAGE_SIZE` was adjusted for 32 MB

this last padding is so that the size of the image is equivalent to the full size of the partitions together. If it does not match up to the full size, the OEM upgrade will not write the last partition.

for 221-B this was 10944k because:
```
(number of blocks in each partition in firmware)
fwconcat0 --> 0x90 --> 144
loader --> 0x01 --> 1
fwconcat1 --> 0x1a --> 26

(64k size of each block)
144 + 1 + 26 = 171
171 * 64k = 10944k
```

this is also the same as the OEM `kernel` + `rootfs`

Reply:
in this case:

```
fwconcat0 --> 0x140
loader --> 0x001
fwconcat1 --> 0x03f
```